kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+import java.nio.ByteBuffer
+import java.util.{ArrayList => JArrayList, Locale}
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+
+import org.apache.hive.service.rpc.thrift._
+
+import org.apache.kyuubi.KyuubiException
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.ProcBuilder
+import org.apache.kyuubi.engine.spark.SparkBatchProcessBuilder
+import org.apache.kyuubi.operation.FetchOrientation.{FETCH_FIRST, FETCH_NEXT, FETCH_PRIOR, FetchOrientation}
+import org.apache.kyuubi.operation.log.OperationLog
+import org.apache.kyuubi.server.api.v1.{BatchRequest, BatchType}
+import org.apache.kyuubi.session.KyuubiBatchSessionImpl
+import org.apache.kyuubi.util.ThreadUtils
+
+class BatchJobSubmission(session: KyuubiBatchSessionImpl, batchRequest: BatchRequest)
+  extends KyuubiOperation(OperationType.UNKNOWN_OPERATION, session) {
```

Comment:
> OperationType.UNKNOWN_OPERATION

I think we can increase the members of OperationType and narrow them to TOperationType

Reply:
But it is difficult to get OperationType from TOperationType.

```
      case TOperationType.UNKNOWN => UNKNOWN_OPERATION
      case other =>
        throw new UnsupportedOperationException(s"Unsupported Operation type: ${other.toString}")
```

Reply:
At last we need reserve that the OperationType of `LaunchEngine` is UNKNOWN_OPERATION.

So, I think we can keep this.

Reply:
OK, I see